ra-tls/sgx-ra-tls/wolfssl-ra-challenger.c

```diff
@@ -297,77 +324,101 @@ int ecdsa_verify_sgx_cert_extensions
-                                                      (char*) evidence.root_ca_crl,
-                                                      (char*) ecdsa_sample_data_real_trustedRootCaCert_pem);
-
-    assert(evidence.qe_identity_len < sizeof(evidence.qe_identity));
-    evidence.qe_identity[evidence.qe_identity_len] = '\0';
-
-    int qe_identity_status = sgxAttestationVerifyQEIdentity((char*) evidence.qe_identity,
-                                                            (char*) evidence.tcb_sign_chain,
-                                                            (char*) evidence.root_ca_crl,
-                                                            (char*) ecdsa_sample_data_real_trustedRootCaCert_pem);
-
-    int quote_status = sgxAttestationVerifyQuote(evidence.quote, evidence.quote_len,
-                                                 (char*) evidence.pck_crt,
-                                                 (char*) evidence.pck_crl,
-                                                 (char*) evidence.tcb_info,
-                                                 (char*) evidence.qe_identity);
-
-    // TODO Update DCAP SDK to newest version.
-    if (quote_status == STATUS_TCB_OUT_OF_DATE) {
-        printf("!!! WARNING: quote verification resulted in STATUS_TCB_OUT_OF_DATE\n");
-        quote_status = 0;
+    int ret = 0;
+    time_t current_time = 0;
+    uint32_t supplemental_data_size = 0;
+    uint8_t *p_supplemental_data = NULL;
+    sgx_status_t sgx_ret = SGX_SUCCESS;
+    quote3_error_t dcap_ret = SGX_QL_ERROR_UNEXPECTED;
+    sgx_ql_qv_result_t quote_verification_result = SGX_QL_QV_RESULT_UNSPECIFIED;
+    uint32_t collateral_expiration_status = 1;
+    quote3_error_t verify_qveid_ret = SGX_QL_ERROR_UNEXPECTED;
+    //specified defined for trusted veirification based on qve.
+    //sgx_enclave_id_t eid = 0;
+    //sgx_launch_token_t token = { 0 };
+    //sgx_ql_qe_report_info_t qve_report_info;
+    //unsigned char rand_nonce[16] = "59jslk201fgjmm;";
+    //int updated = 0;
+
+    //get quote from dcap cert extensions
+    sgx_quote3_t* pquote = NULL;
+    pquote = malloc(8192);
+    if(!pquote){
+       printf("Error: malloc quote buffer failed.\n");
+       return -1;
     }
-
-    int report_user_data_status = verify_report_data_against_server_cert(&crt, (sgx_quote_t*) evidence.quote);
-
-    return pck_crt_status || tcb_info_status || qe_identity_status || quote_status || report_user_data_status;
+    ecdsa_get_quote_from_dcap_cert(der_crt, der_crt_len, pquote);
+    uint32_t quote_size = 436 + pquote->signature_data_len;
+    printf("quote size is %d;  quote signature_data_len is %d.\n", quote_size, pquote->signature_data_len);
+
+    bool verify_by_qve = 0; //1 means trusted verify methond by QvE, 0 means verify by untructed QPL;
+    if(verify_by_qve){ //In current stage, some machine is for pre-prouction and verifying by QvE is not supported;
```

Comment:
Plz fix all this sort of code style. There is a space between "if" and "(" .

Reply:
ok